### PR TITLE
#1772 - fixed: All notifications for Private Group request access are being marked as read once you just visited one of them

### PR DESCRIPTION
**Pull request by milanhirapra**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes https://github.com/buddyboss/buddyboss-platform/issues/1772 .

### How to test the changes in this Pull Request:

1. Click on the Notification icon on the header
2. See the notifications for the group invitation
3. Click on any Group invitation notification then the notification will be marked as read only clicked notification

### Proof Screenshots or Video
https://vimeo.com/582540478/b169acd9af

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-templates/bp-nouveau/js/buddypress-nouveau.js [hunk 1]
```diff
@@ -67,6 +67,9 @@ window.bp = window.bp || {};
 			// Legal agreement enable/disabled submit button.
 			this.enableSubmitOnLegalAgreement();
 
+			// Remove requested param from the URL for Group Invitation
```
Comment (dipskakadiya) [inline src/bp-templates/bp-nouveau/js/buddypress-nouveau.js:70]: @milanhirapra Why do we need to remove `requested` param from JS? Can't we added that conditionally from backend.

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-templates/bp-nouveau/js/buddypress-nouveau.js (hunk 1 above)
Comment (milanhirapra) [reply src/bp-templates/bp-nouveau/js/buddypress-nouveau.js:70]: @dipskakadiya If we click on any notification from the notification list then notification details will appear and at that time this `groups_screen_group_admin_requests` hook is called. So, it's a reason to remove it from the JS.

**Review by dipskakadiya — APPROVED** [review]

**Comment by sumitbuddyboss** [comment]
Hello @milanhirapra 

Issue >
Notification does not disappear after click and also count not upadting.

Issue video:
https://www.loom.com/share/4c3e4c5e365e46cf80aaa052c14530e3

Thank you

**Comment by milanhirapra** [comment]
Hi @mehulkaklotar,

If we click on any group(Group A) request notification then it will redirect to this particular group page(Group A). We can see the user requests on this group page. But, we can't see all group requests(Group A, Group B, Group C) on this page. So, it's an issue and I verified it on our demo site.

Here are the screencasts:
Our Demo site: https://vimeo.com/635929385/b65db690e6
My Local site: https://vimeo.com/635923732/39e84417bd

CC: @sourabhmatolia @chetansatasiya @KartikSuthar

**Comment by milanhirapra** [comment]
Hi @mehulkaklotar,
It's working fine with the `bp_notifications_mark_notifications_by_item_id` function and I made these changes.
Please review it.

**Comment by JetalDobariya** [comment]
Hello @milanhirapra 

I did verify this PR in the QA-Approved branch, now Read-Unread notification is working fine.
--> 
image
image

Thanks!
CC : @sourabhmatolia @mehulkaklotar @KartikSuthar

**Pull request merged**